### PR TITLE
#5649 Fix loop of get capabilites in style editor

### PR DESCRIPTION
**Pull request by MV88**
## Description

The error was not handled when unmashall fails
it had the form of, not a classical request error with status etc
```
"Element [{http://inspire.ec.europa.eu/schemas/inspire_vs/1.0}inspire_vs:ExtendedCapabilities] 
could not be unmarshalled as is not known in this context and the property does not allow DOM content."
```

the fix was in the  web/client/plugins/StyleEditor.jsx how error flag was turned on

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#5649

**What is the new behavior?**

no loop error flow, now it shows
image

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/plugins/StyleEditor.jsx [hunk 1]
```diff
@@ -132,7 +132,7 @@ const StyleEditorPlugin = compose(
                 isEditing: status === 'edit',
                 loading,
                 layer,
-                error: !!(error && error.availableStyles),
+                error: !!(error?.availableStyles || error?.global),
```
Comment (allyoucanmap) [inline web/client/plugins/StyleEditor.jsx:135]: We should specify the error in the message. Currently the title is always 'Missing available styles' and we are adding the error.global as additional flag.

As discussed with @MV88 we should pass the whole error object to the emptyState enhancer and change the content of the message description based on the type of error, something like this:

```js
emptyState(
    ({ error }) => !!(error?.availableStyles || error?.global),
    ({ error }) => {
        if (error.availableStyles) {
            return {....};
        }
         return {....};
    }
),
 
```
we should catch only `error?.availableStyles || error?.global`

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

**Review by allyoucanmap — APPROVED** [review]
@ElenaGallo @tdipisa please on testing phase try also the visual style editor on dev environment

**Comment by tdipisa** [comment]
@ElenaGallo please proceed with tests on DEV as also reported by @allyoucanmap in the above comment. Let us know if we can proceed with the backport.

**Comment by ElenaGallo** [comment]
@tdipisa test done, @MV88 please backport to 2021.01.xx

**Pull request merged**